src/NSubstitute/Core/DefaultForType.cs

```diff
@@ -46,6 +46,15 @@ private object DefaultInstanceOfValueType(Type returnType)
             return BoxedDouble;
         }
 
-        return Activator.CreateInstance(returnType)!;
```

Comment:
I'm doubting if we should use ` Activator.CreateInstance(returnType)!;` for non-stucts, to keep things 100% the same for the previous cases. 

Although all tests works, so not sure.\

@nsubstitute/core-team any opinion on this?

Reply:
So after some reading, this is large change?
```c#
 Activator.CreateInstance(returnType)!;
```
Calls the (default) constructor - that is indeed an issue with structs

```c#
System.Runtime.CompilerServices.RuntimeHelpers.GetUninitializedObject
```

Won't call the constructor. 

I'm not sure if this is a good idea.